### PR TITLE
fix the state traversal for humans

### PR DESCRIPTION
**Pull request by mweiss17**
This PR addresses the issues raised in : https://github.com/pg2455/covid_p2p_simulation/issues/41

Essentially there was a bug that caused humans to not move through the SEIR states. This PR fixes that issue.

### CODE REVIEW

File: simulator.py [hunk 1]
```diff
@@ -267,7 +267,7 @@ def state(self):
         return [int(self.is_susceptible), int(self.is_exposed), int(self.is_infectious), int(self.is_removed)]
 
     def assert_state_changes(self):
-        next_state = {0:1, 1:2, 2:0}
+        next_state = {0:1, 1:2, 2:3}
```
Comment (nasimrahaman) [inline simulator.py:270]: It seems like we're changing a SEIS model to a SEIR model. Is this intended?

File: simulator.py [hunk 2]
```diff
@@ -167,7 +167,7 @@ def is_infectious(self):
 
     @property
     def is_removed(self):
-        return self.recovered_timestamp == datetime.datetime.max
+        return self.recovered_timestamp != datetime.datetime.min
```
Comment (pg2455) [inline simulator.py:170]: when i coded this, i was thinking -

- in the beginning: assign `datetime.datetime.min` to `self.recovered_timestamp`
- (reinfection possible) if someone is infected and has recovered assign `self.env.timestamp` to `self.recovered_timestamp`. This `timestamp` will be used to get time delta between two infections to the same person. 
-  (reinfection possible) if someone is dead assign `datetime.datetime.max` to `self.recovered_timestamp`

So `is_removed` was `self.recovered_timestamp == datetime.datetime.max` (or the agent never recovers).  

The current clause will make the reinfected susceptible agents removed from the population.

File: simulator.py (hunk 1 above)
Comment (pg2455) [inline simulator.py:270]: I was sloppy when I coded this. basically, an infected person can either be dead or be back to susceptible. so the state transition can either be `2 --> 3` or `2 --> 0`. I think we can use a list and assert the membership based on `self.recovered_timestamp`.

File: simulator.py [hunk 3]
```diff
@@ -292,7 +292,7 @@ def run(self, city):
                 assert self.has_logged_symptoms is True # FIXME: assumption might not hold
 
             if self.is_infectious and self.env.timestamp - self.infection_timestamp >= datetime.timedelta(days=self.recovery_days):
-                if self.never_recovers or True: # re-infection assumed negligble
+                if self.never_recovers: # re-infection assumed negligble
```
Comment (pg2455) [inline simulator.py:295]: this assumes re-infection is possible. might be we can add a parameter - REINFECTION_PROBABILITY and use that as a parameter here in `if`?

**Review by pg2455 — COMMENTED** [review]
@mweiss17 Thanks for looking at this issue. I was sloppy while writing the code for `state_transition`, so sorry for that. I have added comments regarding the other choices, and it will be very helpful to clarify them because it might not be a good design. Let me know!

File: simulator.py (hunk 1 above)
Comment (pg2455) [reply simulator.py:270]: It is actually SEIRS. The last two states (RS) transition depend on whether the recovered person transitions to susceptible or is never reinfected/dead (in which case it is just removed).

**Pull request merged**